pkgs/os-specific/linux/rtlwifi_new/default.nix

```diff
@@ -0,0 +1,68 @@
+{ stdenv, lib, fetchFromGitHub, kernel }:
+
+with lib;
+
+let modDestDir = "$out/lib/modules/${kernel.modDirVersion}/kernel/drivers/net/wireless/realtek/rtlwifi";
+
+in
+
+stdenv.mkDerivation rec {
+  name = "rtlwifi_new-git-${version}";
+  version = "2016-09-12";
+
+  src = fetchFromGitHub {
+    owner = "lwfinger";
+    repo = "rtlwifi_new";
+    rev = "7a1b37d2121e8ab1457f002b2729fc23e6ff3e10";
+    sha256 = "0z8grf0fak2ryxwzapp9di77c4bghzkv8lffv76idkcnxgq6sclv";
+  };
+
+  hardeningDisable = [ "pic" "format" ];
+
+  makeFlags = concatStringsSep " " [
+    "KSRC=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build"
+    "MODDESTDIR=$out/lib/modules/${kernel.modDirVersion}/kernel/drivers/net/wireless/realtek/rtlwifi"
+  ];
+
+  enableParallelBuilding = true;
+
+  installPhase = ''
```

Comment:
This could probably need some clean up (at least some consistency whether to prefix with `./` or not and maybe a glob or find could simplify this greatly